### PR TITLE
Feat(rabbithole): initial quest plugin

### PR DESCRIPTION
**Pull request by Quazia**

### CODE REVIEW

File: packages/rabbithole/src/contract-addresses.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+export const RABBITHOLE_QUEST_FACTORY =
```
Comment (mmackz) [inline packages/rabbithole/src/contract-addresses.ts:1]: Keeping the exports below the imports for readability purposes.

File: packages/rabbithole/src/test-transactions.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  v: 1n,
+  value: 0n,
+  typeHex: '0x2' as Hash,
+  yParity: 0,
+} as TransactionEIP1559
+
+export const controlTransaction: TestTransaction = {
+  transaction: CREATE_AND_QUEUE_QUEST_TX,
+  rewardToken: '0x4200000000000000000000000000000000000042',
+  rewardAmount: 1500000000000000000,
```
Comment (mmackz) [inline packages/rabbithole/src/test-transactions.ts:42]: It doesnt appear to be causing any issues, but this number is too large for a standard integer.

**Review by mmackz — APPROVED** [review]
Looks Good!!

**Comment by mmackz** [comment]
added a few tests to make sure undefined parameters don't cause any issues.

File: packages/rabbithole/src/test-transactions.ts (hunk 2 above)
Comment (Quazia) [reply packages/rabbithole/src/test-transactions.ts]: Converted that and a few others to `BigInt` just to be safe 😊 Good catch!

**Pull request merged**